docs/migration-guide-js.md

```diff
@@ -0,0 +1,138 @@
+
+Migration process consists of following steps:
+1. [Considering parsers differences](#considering-parsers-differences)
+2. Installing the new parser (as described in respective [readme section](https://github.com/raml-org/webapi-parser#javascript))
+3. [Migrating the code](#migrating-the-code)
+
+## Considering parsers differences
+There are few differences to consider when migrating to `webapi-parser`:
+* In addition to RAML 0.8 and RAML 1.0 parsing it can also resolve (flatten) it and validate. The parser also supports a number of other API Spec formats: OAS 2.0, OAS 3.0, AMF JSON-LD;
+* `webapi-parser` provides only async/Promise-based API;
+* API of the model/object it produces on parsing is completely different from the one produced by `raml-1-parser`. You can research the new model API by following the link in the [assistance section](#need-assistance) below.
+
+## Migrating the code
+Consider this code which uses `raml-1-parser`:
+```js
+const parser = require('raml-1-parser')
+
+async someFunc () {
+  const api = await parser.loadRAML('/path/to/raml10/api.raml')
+  const expanded = api.expand(true)
+  // Do something with the expanded API
+}
+
+someFunc()
+```
+
+Here's how it can be reworked to use `webapi-parser`:
+```js
+const wap = require('webapi-parser').WebApiParser
+
+async someFunc () {
+  // Or wap.raml08 for RAML 0.8 operations
+  const api = await wap.raml10.parse('file:///path/to/raml10/api.raml')
+  const expanded = await wap.raml10.resolve(api)
+  // Do something with the expanded API
+}
+
+someFunc()
+```
+
+In the example above, object `WebApiParser` contains namespaces for all the supported API Spec formats: `raml10`. `raml08`, `oas20`, `oas30`, `amfGraph`, each having an identical interface (OAS namespaces have an extra few methods). The list of supported operations each format supports includes parsing, resolution(flattening), validation and string generation.
+
+To get a description of each namespace and operation please research the new model API by following the link in the [assistance section](#need-assistance) below.
+
+## Detailed migration examples
+This section lists migration examples of the most common `raml-1-parser` parsing and model methods. Snippets are separated with a newline. First line of each example shows `raml-1-parser` method usage, while the second line shows how to achieve the same functionality with `webapi-parser` if possible. If no obvious alternative exists, a comment gives more detail.
+
+### Parsers
+```js
+const oldParser = require('raml-1-parser')
+const wp = require('webapi-parser')
+const wap = wp.WebApiParser
+
+const node = await oldParser.loadApi('/path/to/api.raml')
+const model = await wap.raml10.parse('file:///path/to/api.raml')
+// or
+// const model = wap.raml08.parse('file:///path/to/api.raml')
+
+oldParser.asFragment(node)
```

Comment:
add a short description of what it does before each block

Reply:
Added